ui/pages/settings/networks-tab/network-tab.js

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  ADD_NETWORK_ROUTE,
+  NETWORKS_FORM_ROUTE,
+} from '../../../helpers/constants/routes';
+import { setSelectedSettingsRpcUrl } from '../../../store/actions';
+import Button from '../../../components/ui/button';
+import { getEnvironmentType } from '../../../../app/scripts/lib/util';
+import { ENVIRONMENT_TYPE_FULLSCREEN } from '../../../../shared/constants/app';
+import {
+  getFrequentRpcListDetail,
+  getNetworksTabSelectedRpcUrl,
+  getProvider,
+} from '../../../selectors';
+import { NETWORK_TYPE_RPC } from '../../../../shared/constants/network';
+import { defaultNetworksData } from './networks-tab.constants';
+import NetworkTabContent from './network-tab-content';
+import NetworkForm from './network-form';
+
+const defaultNetworks = defaultNetworksData.map((network) => ({
+  ...network,
+  viewOnly: true,
+}));
+
```

Comment:
@NiranjanaBinoy : it might be useful to place this in separate file and the component is fairly large.